CombatTimer 2.0.0

iZeaoGamer opened the pull request:
This Pull request pushes out a brand new version, which changes a few aspects of the plugin, to a better state of mind. Though, the plugin’s still the same concept wise, just added a few new features to make the plugin better.

Here are the following updates, made in this newly made v2.0.0 of CombatTimer:

## CombatTimer v2.0.0 CHANGES
- Better formatting for plugin.yml
Implement functional Creative mode checks.
- Rename CreativeCheck() to hasCreativeCheck() for better quality codes. [1 / 2]
- Creative checks no longer depend on the server’s check. It will now require a Player dependency, rather than implementing it to the server, making the plugin non functional. This changes:
1. Parameters for setCreativeCheck(), add’s Player parameter checks.
2. Parameters for hasCreativeCheck(), add’s Player parameter checks.
3. Changed CreativeCheck[] formatted arrays to require player name, instead of being kept false or true without requiring player’s name input.
4. Added onJoin() class event to prevent any future errors with hasCreativeCheck() not being recognised as false or true.
- Reformatted the code to make it look better.
- Fixed Syntax error

This has been untested to work as of yet. But it should work nonetheless. I’ll test this shortly, and get back to you on whether or not this works.

iZeaoGamer committed:
Players can no longer fly whilst in combat.

iZeaoGamer committed:
Better formatting for plugin.yml

iZeaoGamer committed:
Update CombatHUD.php

iZeaoGamer committed:
Update HUDTask.php

iZeaoGamer committed:
Implement functional Creative mode checks.

iZeaoGamer committed:
More updates!

- Rename CreativeCheck() to hasCreativeCheck() for better quality codes. [1 / 2]
- Creative checks no longer depend on the server’s check. It will now require a Player dependency, rather than implementing it to the server, making the plugin non functional. This changes:
1. Parameters for setCreativeCheck(), add’s Player parameter checks.
2. Parameters for hasCreativeCheck(), add’s Player parameter checks.
3. Changed CreativeCheck[] formatted arrays to require player name, instead of being kept false or true without requiring player’s name input.
4. Added onJoin() class event to prevent any future errors with hasCreativeCheck() not being recognised as false or true.

iZeaoGamer committed:
More updates!

- Rename CreativeCheck() to hasCreativeCheck() for better quality codes [2/2]
- Add commented information.

iZeaoGamer committed:
Fix syntax error.

iZeaoGamer committed:
Revert

iZeaoGamer committed:
Fixed syntax error

iZeaoGamer committed:
Rename class to PlayerLoginEvent

This is to prevent undefined index issues.

SahalAmeen reviewed (approved):
I have not tested any of the code. Hope you've done the necessary.

iZeaoGamer committed:
Fixed undefined index issue

This issue was caused because This task wasn’t recognising the player joined. Therefore, it doesn’t recognise the creative check because this task was executed before the join event.
I’ve also added $player->spawned check to ensure the player has spawned into the server. (After loading screen)

iZeaoGamer commented:
Yes, this has been tested to work just now. All good!
